### ISSUE TITLE
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 404: ordinal not in range(128)

### ISSUE THREAD

**Issue by syphernl** [saltstack/salt#47089]
### Description of Issue/Question

After upgrading to Salt 2018.3.0 a minion on 16.04 started having issues with the `ssh` module which is being through `ssh_auth.present`:
```
          ID: my_ssh_key
    Function: ssh_auth.present
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1878, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1823, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/ssh_auth.py", line 340, in present
                  fingerprint_hash_type=fingerprint_hash_type)
                File "/usr/lib/python2.7/dist-packages/salt/modules/ssh.py", line 738, in set_auth_key
                  fingerprint_hash_type=fingerprint_hash_type)
                File "/usr/lib/python2.7/dist-packages/salt/modules/ssh.py", line 476, in check_key
                  fingerprint_hash_type=fingerprint_hash_type)
                File "/usr/lib/python2.7/dist-packages/salt/modules/ssh.py", line 404, in auth_keys
                  keys[u] = _validate_keys(full, fingerprint_hash_type)
                File "/usr/lib/python2.7/dist-packages/salt/modules/ssh.py", line 197, in _validate_keys
                  line += '\n'
              UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 404: ordinal not in range(128)
     Started: 09:44:07.019787
    Duration: 14.039 ms
     Changes:   
```

One of the SSH pubkeys we have contains a diacritic (a `ë` to be exact)

This works fine in `2017.7.4`, but breaks once we upgrade to `2018.3.0`.

`locale` output:
```
root@qa:~# locale
LANG=en_US.UTF-8
LANGUAGE=
LC_CTYPE=en_US.UTF-8
LC_NUMERIC=nl_NL.UTF-8
LC_TIME=nl_NL.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=nl_NL.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=nl_NL.UTF-8
LC_NAME=nl_NL.UTF-8
LC_ADDRESS=nl_NL.UTF-8
LC_TELEPHONE=nl_NL.UTF-8
LC_MEASUREMENT=nl_NL.UTF-8
LC_IDENTIFICATION=nl_NL.UTF-8
LC_ALL=

```

### Versions Report

Master
```
Salt Version:
           Salt: 2018.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: 1.6.1
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.3.7
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-1031-aws
         system: Linux
        version: Ubuntu 16.04 xenial
```

Minion
```
Salt Version:
           Salt: 2018.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.3.7
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.2.3
 
System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-119-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

This also seems to affect `12.04`.

**Comment by gtmanfred** [comment]
Can you run `salt <minionid> test.versions_report` so that we can make sure that the locale data that is being returned is what the daemon is running with, instead of salt-call which would use the locale set in the shell?

Thanks,
Daniel

**Comment by syphernl** [comment]
@gtmanfred as requested:

```
    Salt Version:
               Salt: 2018.3.0
     
    Dependency Versions:
               cffi: Not Installed
           cherrypy: Not Installed
           dateutil: 2.4.2
          docker-py: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.8
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: 1.0.3
       msgpack-pure: Not Installed
     msgpack-python: 0.4.6
       mysql-python: 1.3.7
          pycparser: Not Installed
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
       python-gnupg: Not Installed
             PyYAML: 3.11
              PyZMQ: 15.2.0
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.2.1
                ZMQ: 4.1.4
     
    System Versions:
               dist: Ubuntu 16.04 xenial
             locale: UTF-8
            machine: x86_64
            release: 4.4.0-1054-aws
             system: Linux
            version: Ubuntu 16.04 xenial
```

According to the output, both use `UTF-8`.

We have now worked around this on our end by ensuring our `authorized_keys` do not contain diacritics

**Comment by gtmanfred** [comment]
Ok, looks like this is setup correctly, thanks for reporting.
Daniel

**Comment by rallytime** [comment]
Thanks @syphernl for reporting this. This should be fixed with #47153.

**Comment by rallytime** [comment]
Closed via #47153
